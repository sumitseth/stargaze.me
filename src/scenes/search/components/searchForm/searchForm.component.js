import React from 'react';
import PropTypes from 'prop-types';
import { navigate } from '@reach/router';
import { searchFormTestSelectors } from './searchForm.testSelectors';

const handleOnSubmit = e => {
  e.preventDefault();
  const query = e.target.elements.repos.value.trim();
  navigate(`/search/${query}`);
};

export const SearchForm = ({ onSubmit = handleOnSubmit }) => (
  <form
    data-testid={searchFormTestSelectors.form}
    onSubmit={onSubmit ? e => onSubmit(e) : undefined}
    style={{
      display: 'flex',
      justifyContent: 'center',
      maxWidth: 300,
      margin: 'auto',
    }}
  >
    <input
      type="search"
      name="repos"
      data-testid={searchFormTestSelectors.input}
      placeholder="Enter a Repository name"
      autoFocus
      size={32}
      style={{
        borderRight: 'none',
        borderTopRightRadius: '0',
        borderBottomRightRadius: '0',
      }}
    />
    <button data-testid={searchFormTestSelectors.submit} type="submit">
      Search
    </button>
  </form>
);

SearchForm.propTypes = {
  onSubmit: PropTypes.func,
};

export default SearchForm;
