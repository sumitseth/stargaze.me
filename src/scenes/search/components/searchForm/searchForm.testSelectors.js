export const searchFormTestSelectors = {
  submit: 'search-form-submit',
  form: 'search-form',
  input: 'search-form-input',
};
