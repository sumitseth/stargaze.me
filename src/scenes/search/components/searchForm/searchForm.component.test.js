import React from 'react';
import { render, getByTestId, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SearchForm from './searchForm.component';
import { searchFormTestSelectors } from './searchForm.testSelectors';

it('renders without crashing', () => {
  const { container } = render(<SearchForm />);
  const submit = getByTestId(container, searchFormTestSelectors.submit);
  expect(submit.textContent).toBe('Search');
});

it('calls onSubmit', () => {
  const onSubmit = jest.fn();
  const { container } = render(<SearchForm onSubmit={onSubmit} />);
  const form = getByTestId(container, searchFormTestSelectors.form);
  fireEvent.submit(form);
  expect(onSubmit).toHaveBeenCalled();
});

it('matches snapshot', () => {
  const tree = renderer.create(<SearchForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
