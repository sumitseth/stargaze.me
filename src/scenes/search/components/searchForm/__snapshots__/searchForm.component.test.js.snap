// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  data-testid="search-form"
  onSubmit={[Function]}
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
      "margin": "auto",
      "maxWidth": 300,
    }
  }
>
  <input
    autoFocus={true}
    data-testid="search-form-input"
    name="repos"
    placeholder="Enter a Repository name"
    size={32}
    style={
      Object {
        "borderBottomRightRadius": "0",
        "borderRight": "none",
        "borderTopRightRadius": "0",
      }
    }
    type="search"
  />
  <button
    data-testid="search-form-submit"
    type="submit"
  >
    Search
  </button>
</form>
`;
